### ISSUE TITLE
Support for editing bi-directional relationships with MultiSelectTable

### ISSUE THREAD

**Issue by jgfrm** [viritin/viritin#110]
The current implementation of MultiSelectTable supports the editing of uni-directed JPA relationships.
However, for navigation and other purposes, it is sometimes convenient to use bi-directional relationships.

As the JPA documentation indicates, management of bi-directional relationships is the responsibility of the programmer.
Therefore, some programmer-provided methods should be assumed that manage bidirectional relationships (e.g. if a->b is set, the update method should ensure that b->a is also set).

To this end, the constructor of MultiSelectTable can be changed as follows:

    public MultiSelectTable(String addRemove, String caption)

"addRemove" can then refer to the relation name, e.g. "relatesTo" as used in the JPA class.
 
The JPA class can contain methods such as:
```
     public Set<SomeClass> getRelatesTo() {
     public boolean addRelatesTo(SomeClass obj) // set the relation and inverse relation
     public boolean removeRelatesTo(SomeClass ob) // remove obj from relation and updates inverse relation accordingly
````

The setValue method of MultiSelecTable can then be as follows:
```
     protected void setValue(Object newValue, boolean repaintIsNotNeeded) throws ReadOnlyException {
            Set oldvalue = (Set) getValue();
            super.setValue(newValue, repaintIsNotNeeded);
            if (clientSideChange) {
            	           	
                 Collection collection = getEditedCollection(); 
                Set newvalue = (Set) getValue(); 
                Set orphaned = new HashSet(oldvalue);	
                orphaned.removeAll(newvalue);	

                // remove old values from the JPA set, not being part of the new selection
                removeAll(orphaned);
                
                Set newValues = new LinkedHashSet(newvalue);
                newValues.removeAll(oldvalue);  
                
                // Add the new values
                addAll(newValues);
 
                MultiSelectTable.this.fireValueChange(true);
            }
        }

    };

   private void removeAll(Collection c) {
    	changeAll(c, 0);
    }

    private void addAll(Collection c) {
    	changeAll(c, 1);
    }
    
    private void changeAll(Collection  c, int modus) {
    	
    	String operation = null;
    	
    	if (getEntity() == null) {
    		return;
    	}
    		
    	if (modus == 0) {
    		operation = "removeFrom";
    	} else {
    		if (modus == 1) {
    			operation = "addTo";
    		} else {
    			return;
    		}
    	} 
    	
    	for (Object element : c) {
            	
           Method method = null;         
             try {
                for (Method m : getEntity().getClass().getMethods()) {
                	if ((operation + getAddRemove()).equals(m.getName())) {
                		if (m.getParameterTypes()[0].isAssignableFrom(element.getClass())) {
                			method = m;
                		}
            	     
            	    }
                }
            } catch (SecurityException e) {
            } 
            		
            if (method != null) {
            	try {
             		method.invoke(getEntity(), element);
            	} catch (IllegalArgumentException e) {
            	} catch (IllegalAccessException e) {
            	} catch (InvocationTargetException e) {
            	}
            }
    	}
    }

````

**Comment by mstahv** [comment]
Hi @jgfrm, can you check out the changes I just pushed in. They relate to both this and #111 . Now it should be pretty easy to override couple of methods to update back reference or to do some additional logic.

Let me know what you think about this solution and if it works for you! All enhancement ideas are still welcome.

**Comment by mstahv** [comment]
Hi, I'm closing this for now as it already got released in 1.38. Feel free to reopen or create new issues if this approach is not suitable.
